blog/_posts/2017-06-25-dsl.md

```diff
@@ -0,0 +1,550 @@
++---
+ +layout: post
+ +title:  "Creating domain-specific languages in Julia using macros"
+ +author: <a href="https://github.com/dpsanders">David P. Sanders</a>
+ +---
+# Domain-specific languages in Julia: Generating callable objects using macros
+
+Since the beginning of Julia, it has been tempting to use macros to write **domain-specific languages** (DSLs), i.e. to *extend* Julia syntax to provide a simpler interface to create Julia objects with complicated behaviour. The first, and still most extensive, example is [JuMP](https://github.com/JuliaOpt/JuMP.jl).
+
+Since the fix for the infamous early Julia issue [number #265](https://github.com/JuliaLang/julia/issues/265), which was incorporated in Julia 0.6, released a week ago, some previous methods for creating DSLs in Julia, mainly involving `eval`, ceased to work.
+
+In this post, we will describe a pattern (i.e., a reusable structure) for creating DSLs *without* the use of `eval`; this is the recommended solution.
+
+We will use syntax suitable for Julia 0.6 and later versions. We strongly recommend that everybody upgrade to Julia 0.6.
+
+## Creating a `Model` object containing a function
+
+This blog post arose from a question in the JuliaCon 2017 hackathon about the [Modia modelling language](https://github.com/ModiaSim/Modia.jl), where there is a `@model` macro. Here we will describe the simplest possible version of such a macro, which will create a `Model` object that contains a function, and is itself callable.
+
+First we define the `Model` object. It is tempting to write it like this:
+
```

Comment:
From this point on, spacing and code formatting seem a bit wonky? (Programmatically generated?)